raft/src/main/java/org/apache/kafka/raft/internals/VoterSet.java

```diff
@@ -366,4 +368,16 @@ public static VoterSet fromInetSocketAddresses(String listener, Map<Integer, Ine
 
         return new VoterSet(voterNodes);
     }
+
+    public Optional<Node> idToNode(int id, String listener) {
+        VoterNode voterNode = voters.get(id);
+        if (voterNode == null) {
+            return Optional.empty();
+        }
+        InetSocketAddress address = voterNode.listeners.get(listener);
+        if (address == null) {
+            return Optional.empty();
+        }
+        return Optional.of(new Node(id, address.getHostString(), address.getPort()));
+    }
```

Comment:
Okay. I added a similar method in this PR: https://github.com/apache/kafka/pull/15986/files#diff-7164c449a4cc53dd28cc1a7201fa8b7a824749dab013fb33dff101a1002565daR86-R101

If you agree, do you mind changing the name to `voterNode` to minimize the conflicts. Don't worry about using `ListenerName` instead of `String` as my PR deals with that change.

Reply:
ok